### DISCUSSION TITLE
Best place to keep machine-related data variables when keeping chezmoi.toml on Dropbox?

### DISCUSSION THREAD

**Discussion by halloleo** [twpayne/chezmoi#3830] [Q&A] [answered]
I keep `chezmoi.toml` via a symlink on Dropbox (for all my Internet/Dropbox connected machines).

I still have some variables which are different from machine to machine. What is the best place to keep these variables? 

The `.chezmoidata.toml` file is in the source directory which is synced via the repo. Is there another, external file chezmoi reads?

**Comment by Lockszmith-GH** [comment]
you can use templating with .chezmoidata, reading the values from the system or user input.
https://www.chezmoi.io/user-guide/templating/#using-chezmoitemplates-for-creating-similar-files

**Reply by halloleo** [reply]
Thanks @Lockszmith-GH. I tried using `. chezmoidata.toml` to set machine specific data wioth something oike this:

```
# . chezmoidata.toml\
[data]
keyId = "machine A"
```

but because the file is in the repo *evry* machine got these data, not only machine A. So for me no point to use `.chezmoidata.toml`.

What do I use at the moment? I have some hostname-depended logic in `chezmoi.toml`:

```
[data]
{{- if eq .chezmoi.hostname "mach-1" }}
sysarea = "work"
{{- else if eq .chezmoi.hostname "mach-2" }}
sysarea = "home"
{{- else if eq .chezmoi.hostname "mach-3" }}
sysarea = "home"
{{- else }}
sysarea = ""
{{- end }}
```

But I though it would be nice to separate thsi into different files per machine. - That why I wrote the original post.

**Reply by Lockszmith-GH** [reply]
Maybe use [includeTemplate](https://www.chezmoi.io/reference/templates/functions/includeTemplate/) and use different files based on your host?

**Comment by twpayne** [comment, marked as answer]
You can use different symlinks on each machine.

In your Dropbox folder have one config file per machine. On each machine use a symlink from `~/.config/chezmoi/chezmoi.toml` to the machine-specific config file in your Dropbox folder.
